### ISSUE TITLE
RIO's (<>) conflicts with with Data.Semigroup's

### ISSUE THREAD

**Issue by tejon** [commercialhaskell/rio#33]
I.e., you can't declare a Semigroup instance without hiding `RIO.(<>)`. Not sure how much of an issue this is in practice, but it doesn't seem like it should have to be the case?

**Comment by snoyberg** [comment]
I'm classy-prelude we used <> for semigroup and ++ for Monoid. It works
well in practice IMO, but not sure if that's an acceptable change from
standard practice for rio. On the other hand, with 8.4 coming out soon,
most of the missing instances for Semigroup are being added, so it may not
make much of a difference.

**Comment by snoyberg** [comment]
I pushed through this change in https://github.com/commercialhaskell/rio/commit/cb1cdedac2edce9c19619b433b2709bbc7e0350c#diff-fc787bce63e5df57e918a279f69f25e1L175. If there are still issues with this, please comment, otherwise closing for now.

**Comment by snoyberg** [comment]
Actually, I'm going to revert this and reopen the issue for discussion. Trying out this change in the Stack codebase leads to too much friction with existing libraries that reexport `<>` and `++` from base.

**Comment by tejon** [comment]
Is the friction you mentioned with `<>`, or only with `++`? I'm not sure we need the latter, seeing as Rio is not supporting older versions of base prior to Semigroup becoming a superclass of Monoid. `Semigroup.<>` should always work for both, no?

Granted, I'm completely baffled as to how and why they aren't the same function to begin with, and the answer here probably depends on that. A brief hackage source check tells me nothing; they both point to the same definition in `GHC.Base`.

**Comment by snoyberg** [comment]
>  seeing as Rio is not supporting versions of base prior to Semigroup becoming a superclass of Monoid.

That's not true. We're supporting from GHC 8.0. I think that's where a lot of the misunderstanding going on here is coming from.

**Comment by snoyberg** [comment]
I've opened PR #80 which switches to Semigroup's `<>` usage. This may cause some friction short term, as libraries are still updating to the Semigroup/Monoid switch, but given that 8.4.1 is out the door, that friction should hopefully disappear quickly. To test this out, I've updated Stack's rio branch to use this change:

https://github.com/commercialhaskell/stack/commit/3daafdc2b502f64fb0935bef98a3dcb936bc8554

I think this addresses the original issue here. If so, I think we can merge this and then make a first release of rio.
